api/auth/cert/cert_test.go

```diff
@@ -0,0 +1,213 @@
+package cert
+
+import (
+	"context"
+	"crypto/ecdsa"
+	"crypto/elliptic"
+	"crypto/rand"
+	"crypto/tls"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"encoding/pem"
+	"fmt"
+	"math/big"
+	mathrand "math/rand"
+	"net"
+	"net/http"
+	"os"
+	"path/filepath"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/hashicorp/go-rootcerts"
+	"github.com/hashicorp/vault/api"
+)
+
+func TestLogin(t *testing.T) {
+	certDir := createCertificatePairs(t)
+
+	wg := &sync.WaitGroup{}
+	wg.Add(1)
+	ln, sv := runTestServer(t, certDir, func(w http.ResponseWriter, r *http.Request) {
+		if r.TLS == nil || len(r.TLS.PeerCertificates) == 0 {
+			t.Fatalf("no client cert provided")
+		}
+
+		w.WriteHeader(http.StatusOK)
+		w.Write([]byte(`{"auth": {"client_token": "test-token"}}`))
+
+		wg.Done()
+	})
+	defer sv.Shutdown(context.Background())
+
+	// Create a new CertAuth struct.
+	auth, err := NewCertAuth(
+		"role-name",
+		WithCACert(filepath.Join(certDir, "ca_cert.pem")),
+		WithClientCertAndKey(filepath.Join(certDir, "client_cert.pem"), filepath.Join(certDir, "client_key.pem")),
+	)
+
+	cfg := api.DefaultConfig()
+	cfg.Address = fmt.Sprintf("https://%s", ln.Addr())
+
+	client, err := api.NewClient(cfg)
+	if err != nil {
+		t.Fatalf("failed to create client: %v", err)
+	}
+
+	_, err = auth.Login(context.Background(), client)
+	if err != nil {
+		t.Fatalf("failed to login: %v", err)
+	}
```

Comment:
```suggestion
	secret, err = auth.Login(context.Background(), client)
	if err != nil {
		t.Fatalf("failed to login: %v", err)
	}
	if secret == nil || secret.Auth == nil || secret.Auth.ClientToken != "test-token" {
		t.Fatalf("unexpected token response: %+v", secret)
	}
```
Lets verify that Login returns test-token